File: Data Structures & Algorithms/find-pivot-index/submission-4.py
class Solution:
    def pivotIndex(self, nums: List[int]) -> int:
        prefixSum = []
        for i in range(len(nums)):
            if not prefixSum:
                prefixSum.append(0)
            else:
                prefixSum.append(prefixSum[i-1] + nums[i-1])

        postfixSum = []
        lastSum = 0
        for i in range(len(nums) - 1, -1, -1):
            if not postfixSum:
                postfixSum.append(0)
            else:
                postfixSum.append(lastSum + nums[i+1])
                lastSum = lastSum + nums[i+1]

        print(postfixSum)
        print(prefixSum)

        for i in range(len(nums)):
            if prefixSum[i] == postfixSum[len(nums) -1 - i]:
                return i
        
        return -1
            

File: Data Structures & Algorithms/binary-search/submission-4.cs
public class Solution {
    public int Search(int[] nums, int target) {
        var L = 0;
        var R = nums.Length - 1;

        while (L <= R){
            int mid = L + (R - L) / 2;
            
            if (nums[mid] == target){
                return mid;
            }

            if (nums[mid] > target){
                R = mid - 1;
            }
            else 
            {
                L = mid + 1;
            }
        }
        return -1;
    }
}
